usr/share/rear/format/USB/default/300_format_usb_disk.sh

```diff
@@ -77,14 +80,19 @@ case "$USB_DEVICE_PARTED_LABEL" in
         boot_flag="boot"
         ;;
     "gpt")
-        boot_flag="legacy_boot"
+        # For modern UEFI/GPT booting set ESP boot_flag
+        if [[ -d /sys/firmware/efi/efivars && -f /boot/EFI/systemd/systemd-bootx64.efi || -f /usr/lib/systemd/boot/efi/systemd-bootx64.efi ]]; then
```

Comment:
@ProBackup-nl this is not a standard yet (this path for most common Linux versions)